lib/common/zbuff.h

```diff
@@ -38,7 +38,8 @@ extern "C" {
 /* *************************************
 *  Dependencies
 ***************************************/
-#include <stddef.h>   /* size_t */
+#include <stddef.h>      /* size_t */
+#include "zstd_static.h" /* ZSTD_customMem */
```

Comment:
There is an issue here if `zbuff.h` needs to include `zstd_static.h`.
`zbuff.h` is supposed to be stable and DLL compatible. `zstd_static.h` is clearly not.

On the other hand, `zbuff_static.h` can include `zstd_static.h`.
So I would suggest to move `ZBUFF_createCCtx_advanced()` and `ZBUFF_createDCtx_advanced()` declarations there.